lib/pbench/test/unit/server/query_apis/conftest.py

```diff
@@ -29,17 +29,25 @@ def query_api(
         payload: Dict[str, Any],
         expected_index: str,
         expected_status: str,
+        authorization: bool = True,
         **kwargs,
     ) -> requests.Response:
         host = server_config.get("elasticsearch", "host")
         port = server_config.get("elasticsearch", "port")
         es_url = f"http://{host}:{port}{expected_index}{es_uri}"
         with responses.RequestsMock() as rsp:
             rsp.add(responses.POST, es_url, **kwargs)
-            response = client.post(
-                f"{server_config.rest_uri}{pbench_uri}", json=payload
-            )
-            assert len(rsp.calls) == 1
+            if authorization:
+                response = client.post(
+                    f"{server_config.rest_uri}{pbench_uri}",
+                    headers={"Authorization": "Bearer " + pbench_token},
```

Comment:
You could instead just set `headers = {}` and insert the `authorization` header if necessary, which should allow a single `post` rather than repeating it.